src/libraries/System.IO.Compression/tests/ZipArchive/zip_LargeFiles.cs

```diff
@@ -44,5 +45,81 @@ public static void UnzipOver4GBZipFile()
                 tempDir.Delete(recursive: true);
             }
         }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsSpeedOptimized), nameof(PlatformDetection.Is64BitProcess))] // don't run it on slower runtimes
+        [OuterLoop("It requires 5 GB of free disk space")]
+        public static void CheckZIP64VersionIsSet_ForSmallFilesAfterBigFiles()
+        {
+            // issue #94899
+
+            byte[] largeBuffer = GC.AllocateUninitializedArray<byte>(1_000_000_000); // 1 GB
+            byte[] smallBuffer = GC.AllocateUninitializedArray<byte>(1000);
+
+            string zipArchivePath = Path.Combine(Path.GetTempPath(), "over4GB.zip");
+            DirectoryInfo tempDir = Directory.CreateDirectory(Path.Combine(Path.GetTempPath(), "over4GB"));
+
+            try
+            {
+                using FileStream fs = File.Open(zipArchivePath, FileMode.Create, FileAccess.ReadWrite);
+                const string LargeFileName = "largefile";
+                const string SmallFileName = "smallfile";
+                const uint ZipLocalFileHeader_OffsetToVersionFromHeaderStart = 4;
+                const ushort Zip64Version = 45;
+
+                {
+                    // Create
+
+                    using var archive = new ZipArchive(fs, ZipArchiveMode.Create, true);
+                    ZipArchiveEntry file = archive.CreateEntry(LargeFileName, CompressionLevel.NoCompression);
```

Comment:
is compression enabled an interesting variation?

Reply:
> @karakasa would you mind adding a test that compresses? We just need to make sure the resulting compressed `largefile` should be larger than 4GB too.

Sure. But per my test, the compressed scenario is extremely time-consuming, even for `CompressionLevel.Fastest`.